balboa-core/src/main/scala/com/socrata/balboa/metrics/data/impl/Cassandra11Util.scala

```diff
@@ -109,28 +108,41 @@ object Cassandra11Util extends StrictLogging {
     logger.info("Setting Cassandra socket timeout to '{}'", sotimeout.toString)
     logger.info("Using keyspace '{}'", keyspace)
 
-    val connectionPoolConfiguration = new ConnectionPoolConfigurationImpl("BalboaPool")
-      .setConnectTimeout(sotimeout)
-      .setTimeoutWindow(sotimeout)
-      .setMaxConnsPerHost(connections)
-      .setSeeds(seeds)
 
     // Set local DC as side-effect if specified in configuration.
     // If the local datacenter is specified it will limit the driver
     // to only make connections to the Cassandra nodes in the datacenter
     // and prevent this service from unintentionally reaching across a
     // VPN to connect to a Cassandra node.
-    datacenter.foreach(connectionPoolConfiguration.setLocalDatacenter)
-
-    val cxt:AstyanaxContext[Keyspace] = new Builder()
-      .forKeyspace(keyspace)
-      .withAstyanaxConfiguration(new AstyanaxConfigurationImpl()
-        .setDiscoveryType (NodeDiscoveryType.RING_DESCRIBE)
-      )
-      .withConnectionPoolConfiguration(connectionPoolConfiguration)
-      .withConnectionPoolMonitor(new CountingConnectionPoolMonitor())
-      .buildKeyspace(ThriftFamilyFactory.getInstance())
-    cxt.start()
-    cxt
+
+    val dcPolicy = DCAwareRoundRobinPolicy.builder()
+    datacenter.foreach(dc => dcPolicy.withLocalDc(dc))
+
+    val poolingOptions = new PoolingOptions()
+      .setMaxConnectionsPerHost(HostDistance.LOCAL, connections)
+      .setMaxConnectionsPerHost(HostDistance.REMOTE, connections)
+
+    val socketOptions = new SocketOptions()
+      .setConnectTimeoutMillis(sotimeout)
+
+    val seedInetAddrs = seeds.split(",").map(addrStr => {
+      val addrAndPort = addrStr.trim().split(":")
+      if (addrAndPort.length != 2) {
+        throw new IllegalArgumentException("Address and port must be separated by a ':' in: " + addrStr)
+      }
+      new InetSocketAddress(addrAndPort(0), addrAndPort(1).toInt)
+    })
+
+    logger info s"Connecting to Cassandra on seed addresses: ${ seedInetAddrs.fold("")(_ + ", " + _) }"
```

Comment:
It doesn't seem like this will produce the right output. Won't there will be a leading `,`. Why not use mkString?

    seedInetAddrs.mkString(", ")